docs/user/extension/wmts/index.rst

```diff
@@ -21,8 +15,273 @@ The client code takes care of version negotiation, and even a few server specifi
       <version>${geotools.version}</version>
     </dependency>
 
+WebMapTileServer
+----------------
+
+The client code takes care of version negotiation, and even a few server specific wrinkles for you. It is based on the functionality of AbstractOpenWebService, and handles the requests: GetCapabilities and GetTile.
+
+The client code is fairly slick in that it takes care of version negotiation (inheriting functionalities from AbstractOpenWebService),
+and even a few server specific wrinkles for you.
+
+This page contains examples how how to connect and use the GeoTools WebMapTileServer class. WebMapTileServer acts as a proxy
+for a remote "Web Map Tile Server" and can be used to examine and retrieve published information in the forms of descriptive Java beans, and raw images.
+
+* To begin communicating with a server, pass in a URL pointing to a WMTS.
+
+  Constructing a WebMapTileServer object:
+
+  .. code-block:: java
+
+    URL url = null;
+    try {
+      url = new URL("https://gibs.earthdata.nasa.gov/wmts/epsg4326/best/wmts.cgi");
+    } catch (MalformedURLException e) {
+      // will not happen
+    }
+
+    WebMapTileServer wmts = null;
+    try {
+      wmts = new WebMapTileServer(url);
+    } catch (IOException e) {
+      // There was an error communicating with the server
+      // For example, the server is down
+    } catch (ServiceException e) {
+      // The server returned a ServiceException (unusual in this case)
+    } catch (SAXException e) {
+      // Unable to parse the response from the server
+      // For example, the capabilities it returned was not valid
+    }
```

Comment:
You may wish to pull in a live code example (between two comments) rather than a snippet. 
This ensure the code examples always run; even if the library changes over time.